src/makeInfinityAdder.js

```diff
@@ -4,7 +4,23 @@
  * @return {function}
  */
 function makeInfinityAdder() {
-  // write code here
+  let sum = 0;
+
+  function adder(arg) {
+    if (arg === undefined) {
+      const result = sum;
+
```

Comment:
It might be helpful to add a comment here explaining why the sum is being reset to 0. This is important for understanding the logic of returning the accumulated sum and resetting the state.